libs/remote_storage/src/local_fs.rs

```diff
@@ -343,18 +344,23 @@ impl RemoteStorage for LocalFs {
 
     async fn delete(&self, path: &RemotePath) -> anyhow::Result<()> {
         let file_path = path.with_base(&self.storage_root);
-        if !file_path.exists() {
+        let Ok(metadata) = file_path.metadata() else {
+            // The file either doesn't exist or access is not allowed.
+            // We want to prevent the first situation from yielding an error to mirror S3's behaviour.
             // See https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObject.html
             // > If there isn't a null version, Amazon S3 does not remove any objects but will still respond that the command was successful.
             return Ok(());
```

Comment:
Why allow more than NotFound here?

Since this is a development util, I don't think we should keep accepting any more than the known failure (NotFound) here. Like if you mess up your permissions, better to fail fast.

Reply:
This just mirrors the old code: exists is